samples/PH7/main.c

```diff
@@ -0,0 +1,157 @@
+// This is PH7 29f589677ee88c8958237c4a02d212d59d8b6106 PH7/examples/ph7_intro.c with minor changes
+
+#ifdef NXDK
+#include <hal/video.h>
+#include <hal/debug.h>
+#endif
+/*
+ * This simple program is a quick introduction on how to embed and start
+ * experimenting with the PH7 engine without having to do a lot of tedious
+ * reading and configuration.
+ *
+ * For an introduction to the PH7 C/C++ interface, please refer to this page
+ *        http://ph7.symisc.net/api_intro.html
+ * For the full C/C++ API reference guide, please refer to this page
+ *        http://ph7.symisc.net/c_api.html
+ */
+/*
+ * The following is the PHP program to execute.
+ *   <?php
+ *    echo 'Welcome guest'.PHP_EOL;
+ *    echo 'Current system time is: '.date('Y-m-d H:i:s').PHP_EOL;
+ *    echo 'and you are running '.php_uname();
+ *   ?>
+ * That is, this simple program when running should display a greeting
+ * message, the current system time and the host operating system.
+ * A typical output of this program would look like this:
+ *
+ *	Welcome guest
+ *	Current system time is: 2012-09-14 02:08:44
+ *	and you are running Microsoft Windows 7 localhost 6.1 build 7600 x86
+ *
+ */
+#define PHP_PROG "<?php "\
+                 "echo 'Welcome guest'.PHP_EOL;"\
+                 "echo 'Current system time is: '.date('Y-m-d H:i:s').PHP_EOL;"\
+                 "echo 'and you are running '.php_uname();"\
+                 "?>"
```

Comment:
This should add a final EOL and also sleep for a bit.
On desktop the output looks ugly otherwise. On Xbox it just keeps rebooting way too fast, so the text never really shows up.

Reply:
Things like fatal errors will also *immediately* reboot the Xbox as well.

Reply:
I see this as a small demo, but ideally it would be a lib and people would handle it however they want in their app (either by adding an error handler to their VM, or by hardcoding "good" code like this sample does).